src/core/dev/ib_ctx_handler.h

```diff
@@ -34,106 +34,49 @@
-//
-class ib_ctx_handler : public event_handler_ibverbs {
+class ib_ctx_handler {
 public:
-public:
-    ib_ctx_handler(doca_devinfo *devinfo, const char *ibname, ibv_device *ibvdevice);
+    ib_ctx_handler(const char *ibname);
     virtual ~ib_ctx_handler();
 
-    /*
-     * on init or constructor:
-     *      register to event manager with m_channel and this.
-     * */
-    ibv_pd *get_ibv_pd() { return m_p_ibv_pd; }
-    ibv_device *get_ibv_device() { return m_p_ibv_device; }
-    doca_dev *get_doca_device() const { return m_doca_dev; }
-    doca_flow_port *get_doca_flow_port();
-    doca_flow_pipe *get_doca_root_pipe();
     const std::string &get_ibname() const { return m_ibname; }
-    struct ibv_context *get_ibv_context() { return m_p_ibv_context; }
-    dpcp::adapter *set_dpcp_adapter();
-    dpcp::adapter *get_dpcp_adapter() { return m_p_adapter; }
-    void check_capabilities();
-    void stop_doca_flow_port();
-    xlio_ibv_device_attr *get_ibv_device_attr()
-    {
-        return xlio_get_device_orig_attr(m_p_ibv_device_attr);
-    }
-    xlio_ibv_device_attr_ex *get_ibv_device_attr_ex() { return m_p_ibv_device_attr; }
-    uint32_t mem_reg(void *addr, size_t length, uint64_t access);
-    void mem_dereg(uint32_t lkey);
-    struct ibv_mr *get_mem_reg(uint32_t lkey);
-    uint32_t user_mem_reg(void *addr, size_t length, uint64_t access);
-    bool is_removed() { return m_removed; }
+    bool get_flow_tag_capability() const;
+    bool get_burst_capability() const;
+    bool is_packet_pacing_supported(uint32_t rate) const;
+    void print_val();
+    void convert_hw_time_to_system_time(uint64_t hwtime, struct timespec *systime);
     void set_ctx_time_converter_status(ts_conversion_mode_t conversion_mode);
-    void set_flow_tag_capability(bool flow_tag_capability);
-    bool get_flow_tag_capability() { return m_flow_tag_enabled; } // m_flow_tag_capability
-    void set_burst_capability(bool burst);
-    bool get_burst_capability() { return m_pacing_caps.burst; }
-    bool is_packet_pacing_supported(uint32_t rate = 1);
-    size_t get_on_device_memory_size() { return m_on_device_memory; }
-    bool is_active(int port_num);
-    bool is_mlx4() { return is_mlx4(get_ibname().c_str()); }
-    static bool is_mlx4(const char *dev) { return strncmp(dev, "mlx4", 4) == 0; }
-    virtual void handle_event_ibverbs_cb(void *ev_data, void *ctx);
 
-    void print_val();
+private:
+    std::string m_ibname;
 
-    inline void convert_hw_time_to_system_time(uint64_t hwtime, struct timespec *systime)
+#ifdef DEFINED_DPCP_PATH_RX_OR_TX
+public:
+    ib_ctx_handler_ibv &get_ctx_ibv_dev() { return *m_ctx_ibv_dev; }
+    const ib_ctx_handler_ibv &get_ctx_ibv_dev() const { return *m_ctx_ibv_dev; }
+    void construct_ctx_ibv_dev(ibv_device *ibvdevice)
     {
-        m_p_ctx_time_converter->convert_hw_time_to_system_time(hwtime, systime);
+        m_ctx_ibv_dev = new ib_ctx_handler_ibv(ibvdevice, m_ibname);
```

Comment:
why new and delete and not unique_ptr? it will save some code and be cleaner imho

Reply:
Too much time in XLIO :)

Reply:
Discussed above.